### PR TITLE
Adds pre-commit 

### LINKED ISSUE #7: Setup pre-commit
We decided on using precommit: https://pre-commit.com/
To include:
- ES lint
- detect secrets

### PR DESCRIPTION
**Pull request by BryanGK**
Pre-commit with es-lint and detect secrets. Also includes initial baseline detect secrets file. I'm unsure but I think you'll need to run `pre-commit install` when you pull these changes so your local will know to run the pre-commit.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            "jsx": true
+        },
+        "ecmaVersion": 13,
+        "sourceType": "module"
+    },
+    "plugins": [
+        "react",
+        "@typescript-eslint"
+    ],
+    "rules": {
```
Comment (mikevespi) [inline .eslintrc.js:23]: The idea of having these rules here is to get us to a pass right now, and then remove them and clean up the resulting errors one by one correct? If so can we just toss a comment in there explaining that?

Comment (BryanGK) [reply .eslintrc.js:23]: yes, the intention was to get the check to pass, then we can go back and remove the rule and fix as required.

Comment (wenzowski) [reply .eslintrc.js:23]: In addition to dropping a  comment into this file perhaps we can spawn a new issue where we begin discussing which rules to turn back on? Some of these rules have greater impact than others.
```suggestion
    "rules": {
        # TODO remove rule overrides that affect code quality
```

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  hooks:
+  - id: eslint
+    files: \.[jt]sx?$  # *.js, *.jsx, *.ts and *.tsx
+    types: [file]
+- repo: git@github.com:Yelp/detect-secrets
+  rev: v1.1.0
+  hooks:
+  - id: detect-secrets
+    args: ['--baseline', '.secrets.baseline']
+    exclude: .*/tests/.*
```
Comment (wenzowski) [inline .pre-commit-config.yaml:23]:
```suggestion
    exclude: .*/tests/.*
    
```

Comment (BryanGK) [reply .pre-commit-config.yaml:23]: @wenzowski I'm unsure what you're suggesting here. It appears to be the same.

Comment (BryanGK) [reply .pre-commit-config.yaml:23]: just adding the endline?

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -9,4 +9,5 @@
   hooks:
   - id: detect-secrets
     args: ['--baseline', '.secrets.baseline']
-    exclude: .*/tests/.*
\ No newline at end of file
+    exclude: .*/tests/.*
+
```
Comment (wenzowski) [inline .pre-commit-config.yaml:13]:
```suggestion

```

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  - id: eslint
+    files: \.[jt]sx?$  # *.js, *.jsx, *.ts and *.tsx
+    types: [file]
+- repo: git@github.com:Yelp/detect-secrets
+  rev: v1.1.0
+  hooks:
+  - id: detect-secrets
+    args: ['--baseline', '.secrets.baseline']
+    exclude: .*/tests/.*
+
```
Comment (wenzowski) [inline .pre-commit-config.yaml]:
```suggestion

```
omg whitespace haha

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+- repo: https://github.com/pre-commit/mirrors-eslint
```
Comment (wenzowski) [inline .pre-commit-config.yaml:12]:
```suggestion
- repo: https://github.com/pre-commit/pre-commit-hooks
  rev: v2.3.0
  hooks:
    - id: check-added-large-files
    - id: check-case-conflict
    - id: check-executables-have-shebangs
    - id: check-json
    - id: check-vcs-permalinks
    - id: check-yaml
    - id: end-of-file-fixer
    - id: trailing-whitespace
- repo: https://github.com/pre-commit/mirrors-eslint
```
Addition of these common hooks will resolve many observed git frictions. In particular, the combination of `end-of-file-fixer` and `trailing-whitespace` would resolve the extra chars in the `.pre-commit-config.yaml` itself.

**Review by wenzowski — CHANGES_REQUESTED** [review]

**Comment by wenzowski** [comment]
Should this branch be rebased on `main` to pull in the latest github actions config?

**Comment by BryanGK** [comment]
@wenzowski I rebased from main onto this branch, I think I did it correctly, but not entirely sure

**Comment by wenzowski** [comment]
Hmm, I wonder why `docs/branching_model_team_agreement.md` is showing as added in the diff now. I don't see that locally–I think github's diff engine got confused. The rebase looks good and this branch is now based on 1639afd (current `main`).

**Review by wenzowski — APPROVED** [review]
Confirmed rendering bug on GitHub. This looks ready to go! :shipit:

File: .pre-commit-config.yaml [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    - id: check-yaml
+    - id: end-of-file-fixer
+    - id: trailing-whitespace
+- repo: https://github.com/pre-commit/mirrors-eslint
+  rev: v8.3.0  # Use the sha / tag you want to point at
+  hooks:
+  - id: eslint
+    files: \.[jt]sx?$  # *.js, *.jsx, *.ts and *.tsx
+    types: [file]
+- repo: git@github.com:Yelp/detect-secrets
```
Comment (wenzowski) [inline .pre-commit-config.yaml:18]:
```suggestion
- repo: https://github.com/Yelp/detect-secrets
```
https://results.pre-commit.ci/run/github/409311093/1638483045.Jzwqc6-7TkGXqxfNKxlQuw

**Pull request merged**